test/parallel/test-repl-history-navigation.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+// Flags: --expose-internals
+
+const common = require('../common');
+const stream = require('stream');
+const REPL = require('internal/repl');
+const assert = require('assert');
+
+// Create an input stream specialized for testing an array of actions
+class ActionStream extends stream.Stream {
+  run(data) {
+    const _iter = data[Symbol.iterator]();
+    const doAction = () => {
+      const next = _iter.next();
+      if (next.done) {
+        // Close the repl. Note that it must have a clean prompt to do so.
+        this.emit('keypress', '', { ctrl: true, name: 'd' });
+        return;
+      }
+      const action = next.value;
+
+      if (typeof action === 'object') {
+        this.emit('keypress', '', action);
+      } else {
+        this.emit('data', `${action}`);
+      }
+      setImmediate(doAction);
+    };
+    setImmediate(doAction);
+  }
+  resume() {}
+  pause() {}
+}
+ActionStream.prototype.readable = true;
+
+
+// Mock keys
+const ENTER = { name: 'enter' };
+const UP = { name: 'up' };
+const DOWN = { name: 'down' };
+
+const prompt = '> ';
+
+const tests = [
+  { // creates few history to navigate for
+    env: {},
+    test: [ 'let ab = 45', ENTER,
+            '555 + 909', ENTER,
+            '{key : {key2 :[] }}', ENTER],
+    expected: []
+  },
+  {
+    env: {},
+    test: [UP, UP, UP, DOWN, DOWN, DOWN],
```

Comment:
I suggest to add one more `UP` and one more `DOWN` so we reach the end of the list in both cases.

Reply:
Yeah, taken care of this. Guess should be good now.. cc @BridgeAR